pool/pool.go

```diff
@@ -41,9 +43,9 @@ type Client interface {
 	AnnounceContainerUsedSpace(context.Context, client.AnnounceSpacePrm) (*client.AnnounceSpaceRes, error)
 	EndpointInfo(context.Context, client.EndpointInfoPrm) (*client.EndpointInfoRes, error)
 	NetworkInfo(context.Context, client.NetworkInfoPrm) (*client.NetworkInfoRes, error)
-	PutObject(context.Context, *client.PutObjectParams, ...client.CallOption) (*client.ObjectPutRes, error)
+	ObjectPutInit(context.Context, client.PrmObjectPutInit) (*client.ObjectWriter, error)
```

Comment:
It's just a matter of taste, but if names of other functions of `Client` interface begin with a verb e.g. `DeleteObject`, `HeadObject`, `PutContainer` etc. maybe it'd be more consistent with `PutObjectInit` and `GetObjectInit`?

Reply:
I found it convenient to call the method from general to particular. But I agree that naming should be consistent.

Like - `ObjectGetInit`, dislike - `InitGetObject`/`GetObjectInit` @masterSplinter01 @alexvanin @KirillovDenis @carpawell @fyrchik

Reply:
Brought all methods and types to the same name format.